lib/url.js

```diff
@@ -663,7 +663,7 @@ Url.prototype.resolveObject = function(relative) {
   // then it must NOT get a trailing slash.
   var last = srcPath.slice(-1)[0];
   var hasTrailingSlash = (
-      (result.host || relative.host) && (last === '.' || last === '..') ||
+      (result.host || relative.host || srcPath.length > 1) && (last === '.' || last === '..') ||
```

Comment:
Style: long lines should wrap at 80 columns.  `make jslint` will warn you about that.